### ISSUE TITLE
Exporting an arrow function won't produce documentation

### ISSUE THREAD

**Issue by cullylarson** [esdoc/esdoc#173]
If I have a module like this:

``` js
/**
 * @module formatmicro
 */

/**
 * @param {number} timeMicro
 * @param {object|function} incrementNames
 * @returns {string}
 * @throws {Error} If either argument isn't the correct type
 */
export default (timeMicro, incrementNames) => { }
```

It won't produce any documentation.  However, if I name the exported function, it will:

``` js
/**
 * @module formatmicro
 */

/**
 * @param {number} timeMicro
 * @param {object|function} incrementNames
 * @returns {string}
 * @throws {Error} If either argument isn't the correct type
 */
export default function formatmicro(timeMicro, incrementNames) { }
```

**Comment by h13i32maru** [comment]
@cullylarson Hi! Thanks for this issue. I will fix it in future version.

**Comment by screendriver** [comment]
+1

**Comment by Droow** [comment]
+1

**Comment by Pegase745** [comment]
+1

**Comment by msc654** [comment]
+1

**Comment by henryj** [comment]
+1

**Comment by ipostol** [comment]
+1

**Comment by shfx** [comment]
+1

**Comment by PeteJodo** [comment]
What's required for this to work?

I'm not too familiar with the code and was just taking a very quick glance at it. ESParser should be able to parse it fine.

I assume in [_createDoc](https://github.com/esdoc/esdoc/blob/c7b4d9b415812cf18c451b1a452c9725593d3891/src/Factory/DocFactory.js#L362-L384), it's resulting type is returning as 'Variable' instead of 'Function' but in [_decideType](https://github.com/esdoc/esdoc/blob/c7b4d9b415812cf18c451b1a452c9725593d3891/src/Factory/DocFactory.js#L418-L435) it's also doing a switch that could be because node's `type` was set incorrectly to 'VariableDeclaration' instead of 'FunctionDeclaration'?

**Comment by canastro** [comment]
👍

**Comment by h13i32maru** [comment]
Just yesterday, I released ESDoc v0.5.0. That supports `arrow function` exporting.
Please check it.
